H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md

```diff
@@ -1,576 +1,938 @@
+
+After exploring the results for our classification problem, let’s use AutoML to explore a regression use-case.
+
+
+## Task 5: H2O AutoML Regression
+For our regression use-case, we are using the same dataset and the same training and test sets. But we do need to choose our predictors and response columns, and we will do it as follow:
+
+~~~python
+y_reg = "ORIGINAL_INTEREST_RATE"
+
+ignore_reg = ["ORIGINAL_INTEREST_RATE", 
+              "FIRST_PAYMENT_DATE", 
+              "MATURITY_DATE", 
+              "MORTGAGE_INSURANCE_PERCENTAGE", 
+              "PREPAYMENT_PENALTY_MORTGAGE_FLAG", 
+              "LOAN_SEQUENCE_NUMBER", 
+              "PREPAID", 
+              "DELINQUENT", 
+              "PRODUCT_TYPE"]
+
+x_reg = list(set(train.names) - set(ignore_reg))
+~~~
+ You can print both your y and x variables
+
+~~~python
+print("y:", y_reg, "\nx:", x_reg)
+~~~
+Now we are ready to start our second AutoML model and train it. This time we will use a time constrain, and set `max_runtime_secs` to 900 seconds, or 15 minutes. You will notice that we are specifying the stopping metric and also the sort metric. In the second tutorial, we focused on RMSE and MAE to check the performance of our model, and we noticed that the two values seemed very correlated. For that reason, we could use any of those metrics. We will use the RMSE for early stopping because it penalizes the error more than the MAE, and we will also use it to sort the leaderboard based on the best value.
+
+~~~python
+aml = H2OAutoML(max_runtime_secs = 900,
+                seed = 42,
+                project_name = 'regression',
+                stopping_metric = "RMSE",
+                sort_metric = "RMSE")
+
+%time aml.train(x = x_reg, y = y_reg, training_frame = train)
+~~~
+
+Once it’s done, print the leaderboard.
+
+~~~python
+lb = aml.leaderboard
+lb.head(rows = lb.nrows)
+~~~
+
+![automl-reg-leaderboard_1](assets/automl-reg-leaderboard_1.png)
+![automl-reg-leaderboard_2](assets/automl-reg-leaderboard_2.png)
+
+
+The leaderboard shows that the GBM models clearly dominated this task. We can retrieve the best model with the `model.leader` command; but, what if we wanted to get another model from our leaderboard? One way to do so is shown below:
+
+~~~python
+# Get model ids for all models in the AutoML Leaderboard
+model_ids = list(aml.leaderboard['model_id'].as_data_frame().iloc[:,0])
+
+# Get the top GBM model
+gbm = h2o.get_model([mid for mid in model_ids if "GBM" in mid][0])
+~~~
+Note that you would need to change the name `GBM`, in the code above, to the name of the model that you want, and that should retrieve the desired model. For example, if you wanted to get the best XGBoost in the leaderboard, you would need to make the following change (you do not need to run the following line of code, this is just an example)
```

Comment:
rewrite the following sentence is a bit confusing: "Note that you would need to change the name `GBM`, in the code above, to the name of the model that you want, and that should retrieve the desired model."